hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleContextImpl.java

```diff
@@ -290,7 +300,7 @@ public PreHandleContext requireKeyOrThrow(
         }
 
         // Verify this key isn't for an immutable account
-        verifyIsNotImmutableAccount(key, responseCode);
+        verifyNotStakingAccounts(account.accountIdOrThrow(), responseCode);
```

Comment:
Now that you have correct checks for hollow accounts (🙌), can we do this check **only** inside the `if (isHollow(account))` blocks? (Since any non-hollow account is definitely not `0.0.800` or `0.0.801`.)

Reply:
Agreed 🙌 . Fixed